test-rpc-functions/src/rpc.rs

```diff
@@ -0,0 +1,207 @@
+    ) -> rpc::Result<bool> {
+        let public_key = crypto::key::PublicKey::hex_decode_all(public_key_hex)
+            .map_err(rpc::Error::to_call_error)?;
+
+        let message = Vec::<u8>::hex_decode_all(message_hex).map_err(rpc::Error::to_call_error)?;
+        let signature =
+            Signature::hex_decode_all(signature_hex).map_err(rpc::Error::to_call_error)?;
+
+        let verify_result = public_key.verify_message(&signature, &message);
+
+        Ok(verify_result)
+    }
+
+    async fn new_vrf_private_key(&self) -> rpc::Result<String> {
+        let keys =
+            crypto::vrf::VRFPrivateKey::new_from_entropy(crypto::vrf::VRFKeyKind::Schnorrkel);
+
+        Ok(keys.0.hex_encode())
+    }
+
+    async fn vrf_public_key_from_private_key(
+        &self,
+        private_key_hex: String,
+    ) -> rpc::Result<String> {
+        let private_key = crypto::vrf::VRFPrivateKey::hex_decode_all(private_key_hex)
+            .map_err(rpc::Error::to_call_error)?;
+
+        let public_key = crypto::vrf::VRFPublicKey::from_private_key(&private_key);
+
+        Ok(public_key.hex_encode())
+    }
+
+    async fn sign_message_with_vrf_private_key(
+        &self,
+        private_key_hex: String,
+        epoch_index: EpochIndex,
+        random_seed: String,
+        block_timestamp: u64,
+    ) -> rpc::Result<String> {
+        let random_seed: H256 =
+            H256::hex_decode_all(random_seed).map_err(rpc::Error::to_call_error)?;
+
+        let transcript = construct_transcript(
+            epoch_index,
+            &random_seed,
+            BlockTimestamp::from_int_seconds(block_timestamp),
+        );
+        let private_key = crypto::vrf::VRFPrivateKey::hex_decode_all(private_key_hex)
+            .map_err(rpc::Error::to_call_error)?;
+
+        let signature = private_key.produce_vrf_data(transcript.into());
+
+        Ok(signature.hex_encode())
+    }
+
+    async fn verify_then_get_vrf_output(
+        &self,
+        epoch_index: EpochIndex,
+        random_seed: String,
+        vrf_data: String,
+        vrf_public_key: String,
+        block_timestamp: BlockTimestamp,
+    ) -> rpc::Result<String> {
+        let vrf_data =
+            crypto::vrf::VRFReturn::hex_decode_all(vrf_data).map_err(rpc::Error::to_call_error)?;
+        let vrf_public_key = crypto::vrf::VRFPublicKey::hex_decode_all(vrf_public_key)
+            .map_err(rpc::Error::to_call_error)?;
+        let random_seed = H256::hex_decode_all(random_seed).map_err(rpc::Error::to_call_error)?;
+
+        let vrf_output = verify_vrf_and_get_vrf_output(
+            epoch_index,
+            &random_seed,
+            &vrf_data,
+            &vrf_public_key,
+            block_timestamp,
+        )
+        .map_err(RpcTestFunctionsError::from);
+
+        let vrf_output = handle_error(vrf_output)?;
+
+        Ok(vrf_output.hex_encode())
+    }
+}
+
+// fn handle_call_error<T>(e: Result<Result<T, RpcTestFunctionsError>, CallError>) -> rpc::Result<T> {
```

Comment:
commented code

Reply:
Fixed.

Reply:
Cool, thanks. Once @azarovh finishes his review (and of the updated fixes), I'll merge and push tests to a new PR.